benchmark/benchmarks.jl

```diff
@@ -0,0 +1,93 @@
+
+SUITE["operations"]["tensor"] = BenchmarkGroup()
+SUITE["operations"]["tensor"]["ket"] = @benchmarkable $k1 ⊗ $k2
+SUITE["operations"]["tensor"]["op"] = @benchmarkable $A ⊗ $B
+SUITE["operations"]["tensor"]["many"] = @benchmarkable $A ⊗ $B ⊗ $C ⊗ $A ⊗ $B ⊗ $C
+
+# Linear algebra operations
+SUITE["linalg"] = BenchmarkGroup(["symbolic"])
+SUITE["linalg"]["trace"] = @benchmarkable tr($A)
+SUITE["linalg"]["ptrace"] = @benchmarkable ptrace($A ⊗ $B, 1)
+SUITE["linalg"]["inverse"] = @benchmarkable inv($A)
+SUITE["linalg"]["dagger"] = @benchmarkable dagger($A)
+SUITE["linalg"]["conjugate"] = @benchmarkable conj($A)
+SUITE["linalg"]["transpose"] = @benchmarkable transpose($A)
+SUITE["linalg"]["commutator"] = @benchmarkable commutator($A, $B)
+SUITE["linalg"]["anticommutator"] = @benchmarkable anticommutator($A, $B)
+
+# Simplification benchmarks
+SUITE["manipulation"] = BenchmarkGroup(["symbolic"])
+compact_pauli_expr = (H * (X+Y+Z)) * (H * (Y1+Y2+Z1))
+expanded_pauli_expr = qexpand(compact_pauli_expr)
+commutator_expr = commutator(X, commutator(Y, commutator(X, Z)))
+
+SUITE["manipulation"]["expand"]["distribution"] = @benchmarkable qexpand($compact_pauli_expr)
+SUITE["manipulation"]["expand"]["commutator"] = @benchmarkable qexpand($commutator_expr)
+SUITE["manipulation"]["simplify"]["applicable_rules"] = @benchmarkable qsimplify($expanded_pauli_expr, rewriter=qsimplify_pauli)
+# Expression isn't simplified with fock rules. Testing how fast we go through inapplicable rules.
+SUITE["manipulation"]["simplify"]["irrelevant_rules"] = @benchmarkable qsimplify($expanded_pauli_expr, rewriter=qsimplify_fock)
+SUITE["manipulation"]["simplify"]["commutator"] = @benchmarkable qsimplify($commutator_expr, rewriter=qsimplify_commutator)
+
+# Expression benchmarks
+SUITE["express"] = BenchmarkGroup(["express"])
+clear_cache!(x) = empty!(x.metadata.express_cache)
+simple_op = X⊗Y
+simple_ket = Z1⊗Z1
+pauli_op_4 = YCX * transpose(Z⊗Y) * YCZ * dagger(Y⊗X) * conj(ZCY) * exp(X ⊗ (2-3im)X) 
+pauli_state_8 = (conj(XCY⊗Y) + dagger(Z⊗YCX) + transpose(0.5im*Y ⊗ exp(XCY))) * (X1⊗Y1⊗Z1)
+
+SUITE["express"]["optics"]["simple_op"] = @benchmarkable express($simple_op) setup=clear_cache!(simple_op) evals=1
+SUITE["express"]["optics"]["simple_ket"] = @benchmarkable express($simple_ket) setup=clear_cache!(simple_ket) evals=1
+SUITE["express"]["optics"]["pauli_op_4"] = @benchmarkable express($pauli_op_4) setup=clear_cache!(pauli_op_4) evals=1
+SUITE["express"]["optics"]["pauli_state_8"] = @benchmarkable express($pauli_state_8) setup=clear_cache!(pauli_state_8) evals=1
+
+# TODO: not sure what else to add here
+SUITE["express"]["clifford"]["simple_ket"] = @benchmarkable express($simple_ket, CliffordRepr()) setup=clear_cache!(simple_ket) evals=1
+SUITE["express"]["clifford"]["simple_observable"] = @benchmarkable express($simple_op, CliffordRepr(), UseAsObservable()) setup=clear_cache!(simple_op) evals=1
```

Comment:
What I had the most trouble with is expressing in Clifford formalism as it seems that most operations are not supported except for the most basic tensor products.

Reply:
I think we can just stick to what you already have here. Expanding on the Clifford capabilities would need to be a separate task.